keccak-hash/src/lib.rs

```diff
@@ -45,13 +45,57 @@ pub fn keccak256(data: &mut [u8]) {
 	keccak256.finalize(data);
 }
 
+/// Computes in-place keccak256 hash of `data[range]`.
+/// # Panics
+///
+/// If `range` is out of bounds.
+///
+/// # Example
+///
+/// ```
+/// let mut data = [1u8; 32];
+/// keccak_hash::keccak256_range(&mut data, 0..8);
```

Comment:
I would expect this to give me the keccak hash back for the first 8 bytes of `data`, is that how this works? The example and docs are not clear.
How does `let mut data = [1u16; 32]; keccak256_range(&mut data, 0..2)` work? Does it act on the two first `u16`s or on the two first bytes? I'd expect the former but I suspect it's the latter.

Reply:
not sure what you mean, data is always `&mut [u8]`, it can't be `&mut [u16]`
so it always operates on bytes

Reply:
Oh, good pt, apologies.

The rest of the comment still stands though: please document the method for dummies (I'm your guinea pig).

Reply:
Added more docs, please take another look. Suggestions are always welcome :)